custom_components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,545 @@
+
+            current_mode = self._mode().value_selected
+            _hvac_temp = next(
+                (
+                    key
+                    for key, value in self._hvac_mapping.items()
+                    if value == current_mode
+                ),
+                None,
+            )
+
+            if _hvac_temp is None:
+                # The current mode is not a hvac mode
+                if (
+                    "heat" in current_mode.lower()
+                    and HVAC_MODE_HEAT in self._hvac_mapping.keys()
+                ):
+                    # The current preset modes maps to HVAC_MODE_HEAT
+                    self._hvac_mode = HVAC_MODE_HEAT
+                elif (
+                    "cool" in current_mode.lower()
+                    and HVAC_MODE_COOL in self._hvac_mapping.keys()
+                ):
+                    # The current preset modes maps to HVAC_MODE_COOL
+                    self._hvac_mode = HVAC_MODE_COOL
+                else:
+                    # The current preset modes maps to self._default_hvac_mode
+                    self._hvac_mode = self._default_hvac_mode
+                self._preset_mode = next(
+                    (
+                        key
+                        for key, value in self._preset_mapping.items()
+                        if value == current_mode
+                    ),
+                    current_mode,
+                )
+            else:
+                # The current mode is a hvac mode
+                self._hvac_mode = _hvac_temp
+                self._preset_mode = PRESET_NONE
+
+    def _update_current_temp(self):
+        """Update current temperature."""
+        if self.values.temperature:
+            self._current_temperature = self.values.temperature.value
+            device_unit = self.values.temperature.units
+            if device_unit is not None:
+                self._unit = device_unit
+
+    def _update_fan_mode(self):
+        """Update fan mode."""
+        if self.values.fan_mode:
+            self._current_fan_mode = self.values.fan_mode.value_selected
+            fan_modes = self.values.fan_mode.value_list
+            if fan_modes:
+                self._fan_modes = list(fan_modes)
+
+    def _update_swing_mode(self):
+        """Update swing mode."""
+        if self._zxt_120 == 1:
+            if self.values.zxt_120_swing_mode:
+                self._current_swing_mode = self.values.zxt_120_swing_mode.value
+                swing_modes = self.values.zxt_120_swing_mode.value_list
+                if swing_modes:
+                    self._swing_modes = list(swing_modes)
+
+    def _update_target_temp(self):
+        """Update target temperature."""
+        current_setpoints = self._current_mode_setpoints()
+        self._target_temperature = None
+        self._target_temperature_range = (None, None)
+        if len(current_setpoints) == 1:
+            (setpoint,) = current_setpoints
```

Comment:
To keep it more readable. 

```suggestion
            setpoint = current_setpoints[0]
```